insights/parsers/zdump_v.py

```diff
@@ -0,0 +1,103 @@
+
+The ``/usr/sbin/zdump -v /etc/localtime -c 2019,2039`` command provides information about
+'Daylight Saving Time' in file /etc/localtime from 2019 to 2039.
+
+Sample content from command ``zdump -v /etc/localtime -c 2019,2039`` is::
+    /etc/localtime  Sun Mar 10 06:59:59 2019 UTC = Sun Mar 10 01:59:59 2019 EST isdst=0 gmtoff=-18000
+    /etc/localtime  Sun Mar 10 07:00:00 2019 UTC = Sun Mar 10 03:00:00 2019 EDT isdst=1 gmtoff=-14400
+    /etc/localtime  Sun Nov  7 05:59:59 2038 UTC = Sun Nov  7 01:59:59 2038 EDT isdst=1 gmtoff=-14400
+    /etc/localtime  Sun Nov  7 06:00:00 2038 UTC = Sun Nov  7 01:00:00 2038 EST isdst=0 gmtoff=-18000
+
+Examples:
+    >>> dst = zdump.data[0]
+    >>> dst.get('utc_time')
+    datetime.datetime(2019, 3, 10, 6, 59, 59)
+    >>> dst.get('utc_time_s')
+    'Sun Mar 10 06:59:59 2019 UTC'
+    >>> dst.get('local_time')
+    datetime.datetime(2019, 3, 10, 1, 59, 59)
+    >>> dst.get('local_time_s')
+    'Sun Mar 10 01:59:59 2019 EST'
+    >>> dst.get('isdst')
+    False
+    >>> dst.get('gmtoff')
+    -18000
+"""
+
+from datetime import datetime
+from insights.specs import Specs
+from insights.parsers import SkipException
+from insights import parser, CommandParser
+
+
+def str2datetime(timestamp):
+    # remove redundant white spaces
+    # fs stands for 'fields'
+    fs = [s for s in timestamp.split(' ') if len(s) > 0]
+    if len(fs) != 6:
+        return
+
+    # Formatting the output
+    # Can't use `" ".join(fs)` for the Fixed width number of timestamp
+    time_string = "%s %s %2s %s %s %s" % (fs[0], fs[1], fs[2], fs[3], fs[4], fs[5])
+
+    # Fixed the problem that the program running this python code doesn't
+    # has the corresponding TimeZone where strptime will raise ValueError.
+    if 'UTC' in fs:
+        time_s = time_string
+        time_f = "%a %b %d %H:%M:%S %Y %Z"
+    else:
+        # Skip the `TimeZone`
+        time_s = "%s %s %2s %s %s" % (fs[0], fs[1], fs[2], fs[3], fs[4])
+        time_f = "%a %b %d %H:%M:%S %Y"
+
+    try:
+        time = datetime.strptime(time_s, time_f)
+    except ValueError:
+        pass
+
+    return time, time_string
+
+
+@parser(Specs.zdump_v)
+class ZdumpV(CommandParser):
+    """Class for command: /usr/sbin/zdump -v /etc/localtime -c 2019,2039"""
+    def parse_content(self, content):
+        if not content:
+            raise SkipException("No Data from command: /usr/sbin/zdump -v /etc/localtime -c 2019,2039")
+
+        self._data = []
+        for line in content:
+            dst = {}
+            if 'UTC' not in line or 'isdst' not in line:
+                # skip the line that does not include a timestamp
+                continue
+
+            utc_time, remains = line.strip('/etc/localtime').split(' = ')
+            dst['utc_time'], dst['utc_time_s'] = str2datetime(utc_time)
+            if dst['utc_time'] is None:
+                continue
+
+            local_time, _ = remains.split("isdst")
```

Comment:
How about

```
utc_time, remains = line.strip('/etc/localtime').split(' = ')
local_time, remains = [i.strip() for i in remains.split("isdst", 1)]
isdst, remains = [i.strip('=') for i in remains.split(None, 1)
gmtoff = remains.split('=')[1]
# code to combine the namedtuple 
```

Reply:
If we write it in the way and command  `zdump` changes the order in the feature :
from:  Sun Mar 10 01:59:59 2019 EST `isdst=0 gmtoff=-18000`
to:      Sun Mar 10 01:59:59 2019 EST `gmtoff=-18000 isdst=0`

We need to modify our code.

Reply:
`gmtoff ` isn't documented in the Man Page. It's an option.

Reply:
Then you can add some safety check, what I gave is just a prototype.